apps/emqx_authn/src/emqx_authn_api.erl

```diff
@@ -1006,11 +1006,10 @@ update_authenticator(ConfKeyPath, ChainName, AuthenticatorID, Config) ->
         )
     of
         {ok, #{
-            post_config_update := #{emqx_authentication := #{id := ID}},
-            raw_config := AuthenticatorsConfig
+            post_config_update := #{emqx_authentication := #{id := _ID}},
+            raw_config := _AuthenticatorsConfig
         }} ->
-            {ok, AuthenticatorConfig} = find_config(ID, AuthenticatorsConfig),
-            {200, maps:put(id, ID, convert_certs(fill_defaults(AuthenticatorConfig)))};
```

Comment:
I'm afraid this has to be reverted.

Reply:
this is not an echo, the `convert_certs` will store files to disk, and return stored paths.
and `fill_defaults` will fill default values for the missing fields in the request.

Reply:
I've been reading through the code a bit now. `convert_certs` does have side-effects, it *reads* `pem` files and checks if they are valid. But it doesn't look like it's creating anything anywhere. 

But `update_config` might trigger creating of ssl related files. Couldn't quite trace it back but there is `emqx_authentication_config:pre_config_update` which later calls some other `convert_certs` within its own module. And this one, deep down the line does a call to `save_pem_file` eventually. 

I just assume that through some callbacks `update_config` does call `emqx_authentication_config:pre_config_update` and as such create those `pem` file(s). Means it should happen with or without that call to that *local* `convert_certs`. 

It still might be necessary to do call those checks to trigger a crash just to indicate there's something wrong, but in that case a proper error handling would we preferable. And it should already happen within `update_config` I'd think.